parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestEncryptionOptions.java

```diff
@@ -125,150 +133,337 @@
+  private String getFileName(EncryptionConfiguration encryptionConfiguration) {
+    return encryptionConfiguration.toString().toLowerCase() + ".parquet.encrypted";
+  }
+
+  private void testReadEncryptedParquetFiles(Path root, List<SingleRow> data) {
     Configuration conf = new Configuration();
+    DecryptionConfiguration[] decryptionConfigurations = DecryptionConfiguration.values();
+    for (DecryptionConfiguration decryptionConfiguration : decryptionConfigurations) {
+      EncryptionConfiguration[] encryptionConfigurations = EncryptionConfiguration.values();
+      for (EncryptionConfiguration encryptionConfiguration : encryptionConfigurations) {
+        Path file = new Path(root, getFileName(encryptionConfiguration));
+        LOG.info("==> Decryption configuration {}", decryptionConfiguration);
+        FileDecryptionProperties fileDecryptionProperties = decryptionConfiguration.getDecryptionProperties();
 
-    for (Map.Entry<DecryptionConfiguration, FileDecryptionProperties> decryptionConfigurationEntry : decryptionPropertiesMap.entrySet()) {
-      DecryptionConfiguration decryptionConfiguration = decryptionConfigurationEntry.getKey();
-      LOG.info("==> Decryption configuration {}", decryptionConfiguration);
-      FileDecryptionProperties fileDecryptionProperties = decryptionConfigurationEntry.getValue();
+        LOG.info("--> Read file {} {}", file.toString(), encryptionConfiguration);
 
-      File folder = new File(root.toString());
-      File[] listOfFiles = folder.listFiles();
+        // Read only the non-encrypted columns
+        if ((decryptionConfiguration == DecryptionConfiguration.NO_DECRYPTION) &&
+          (encryptionConfiguration == EncryptionConfiguration.ENCRYPT_COLUMNS_PLAINTEXT_FOOTER)) {
+          conf.set("parquet.read.schema", Types.buildMessage()
+            .optional(INT32).named(PLAINTEXT_INT32_FIELD_NAME)
+            .named("FormatTestObject").toString());
+        }
 
-      for (int fileNum = 0; fileNum < listOfFiles.length; fileNum++) {
-        Path file = new Path(listOfFiles[fileNum].getAbsolutePath());
-        if (!file.getName().endsWith("parquet.encrypted")) { // Skip non encrypted files
-          continue;
+        int rowNum = 0;
+        try (ParquetReader<Group> reader = ParquetReader.builder(new GroupReadSupport(), file)
+          .withConf(conf)
+          .withDecryption(fileDecryptionProperties)
+          .build()) {
+          for (Group group = reader.read(); group != null; group = reader.read()) {
+            SingleRow rowExpected = data.get(rowNum++);
+            // plaintext columns
+            if ((null != rowExpected.plaintext_int32_field) &&
+              rowExpected.plaintext_int32_field != group.getInteger(PLAINTEXT_INT32_FIELD_NAME, 0)) {
+              addErrorToErrorCollectorAndLog("Wrong int", encryptionConfiguration, decryptionConfiguration);
+            }
```

Comment:
What do you think about this way?

```
import static org.hamcrest.CoreMatchers.equalTo;
// ...
  if (null != rowExpected.plaintext_int32_field)
    errorCollector.checkThat("Wrong int", group.getInteger(PLAINTEXT_INT32_FIELD_NAME, 0), equalTo(rowExpected.plaintext_int32_field));
  }
```
Similarly, you may do the check like that in any other places just like you would do with `assertEquals`. (You may find a couple of other _matchers_ as well.)
UPDATE: Meanwhile, I realized that you are not logging the values but the crypto configurations which are more relevant for this test. So, you may either skip my comment above or use it for creating another `addErrorToCollectorAndLog` like method.

BTW, why do we need to check the expected value for `null`? Shouldn't be the actual value `null` as well?

Reply:
@gszadovszky, If you don't mind, I'd rather leave the method as it is, since it both adds an error to errorCollector for the final result and logs it for a more readable test progress log. If I let errorCollector.checkThat() do the comparison, I will need the current check anyway in order to add the test progress logging.

Thanks, I've removed the null check.